api/anchor.capnp

```diff
@@ -24,8 +24,86 @@ interface Anchor {
     # anchors in the path that do not currently exist are created along
     # the way.
 
+    cell @2 () -> (loader :Loader, storer :Storer, swapper :Swapper);
+    # Cell returns a set of capabilities that provide access to a boxed
+    # value, assigned to the Anchor. Zero or more of these capabilities
+    # MAY be witheld, in which case the corresponding return value will
+    # be null. The loader and storer capabilities respectively map onto
+    # read and write permissions. The swapper is an extended read/write
+    # capability, which SHALL be witheld if either loader or storer are
+    # witheld.
+
     struct Child {
```

Comment:
I agree that ideally we'd add a `name` field to an `Anchor` and call it a day, but unfortunately we can't do that.  We could add a `name` _method_ to the `Anchor` interface, but then fetching the name would require an additional network round-trip 😕

One alternative would be this:

```capnp
ls @0 () -> (children :List(Anchor), names :List(Text));
```

Instead of returning a list of `(Anchor, Text)` tuples, we'd return a list of anchors and a list of names.  This would add 4 bytes of overhead for the length of the second list, so definitely not a show-stopper, though technically rudundant.

Another alternative would be to make `Anchor` a struct type that holds a walk/ls capability.  Something like this:

```capnp
struct Anchor {
    name @0 :Text;
    tree    @1 :Tree;  # name chosen arbitrarily, for illustration...

    interface Text {
        ls @0 () -> (children :List(Anchor));
        walk @1 (path :Text) -> (anchor :Anchor);
    }
}
```
Personally, I find this a bit more convoluted, though.

Reply:
BTW, this would be a good time for us to tweak the `Anchor` schema, so I'm definitely open to discussing this further.